public/src/pages/TeacherPage.tsx

```diff
@@ -154,6 +157,37 @@ export class TeacherPage extends ViewPage {
         return <div>404 Page not found</div>;
     }
 
+    public async groupresults(info: INavInfo<{ course: string }>): View {
+        const courseId = parseInt(info.params.course, 10);
+        const course = await this.courseMan.getCourse(courseId);
+        if (course) {
+            var linkedGroups: IGroupCourseWithGroup[] = [];
+            const groupCourses = await this.courseMan.getCourseGroups(course.id);
+            for (const grpCourse of groupCourses) {
+                const grp = await this.courseMan.getGroupCourse(grpCourse, course);
+                if (grpCourse && grp) {
+                    linkedGroups.push({
+                        course: grp,
+                        group: grpCourse,
+                    
+                    });
+                }
+            }            
+            const labs: IAssignment[] = await this.courseMan.getAssignments(courseId);
+            return <GroupResults 
+                    course={course} 
+                    labs={labs} 
+                    groups={linkedGroups}
+                    onApproveClick={async (submissionID:number) => {
+                        await this.courseMan.approveSubmission(submissionID);
+                        this.navMan.refresh();
+                    }}
+                    >
+                </GroupResults>;
+        }
+        return <div>404 Page not found</div>;
```

Comment:
Suggest to make a `PageNotFound` a global constant and use everywhere where such `<div>`s are being used. Or perhaps better make it a function that takes a specific error message as input and produces a debuggable error message. That is, care should be taken to avoid leaking sensitive information through error messages.

Reply:
Yes, debuggable error messages or for production a more suitable error message would be preferable, I do not know how much time this will take to implement, so in the meantime I can make a global constant.